packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -724,30 +720,168 @@ class TimeInput extends React.Component {
+        best we can do is say "hour" and we have this scenario:
+
+          ==SCREEN READER'S LIST OF FORM INPUTS TO PICK==
+          1. Hour
+          2. Minute
+          3. Second
+          4. Hour <-- Confusing which hour/minute is which because there is no label.
+          5. Minute
+
+        The AccessibleValue's job is to announce a localized human-readable representation of the time to screen reader
+        users. The announcement is only made when the value changes to a valid time. If the Time of Birth started out at
+        T14:22 and the user changed the hour from 02 to 03, the value would be T15:22 and the announcer would
+        immediately say "Time of birth 3:22 pm." or just "3:22 pm" if no label is provided.
+
+        It's important to give screen reader users an indication if the Time Input is disabled, incomplete or invalid
+        just as we do for visual users. Since the isInvalid and required flags are component-wide but
+        aria-invalid/required are for inputs, we must mark all inputs the same. It's not perfect but it gives better
+        information than nothing.
+
+        Unanswered questions:
+        1 - why does it mean to have an invalid meridiem? Why is that possible? It is not supported by Aria or HTML to indicate a button is "invalid".
+        2 - why won't the mobile invalid and mobile incomplete tests read the inputs as invalid or incomplete? The voiceOver rotor and the accessibility panel both show them correctly as incomplete or invalid, just the read-mode is wrong. This problem doesn't happen on the normal incomplete/invalid tests. I was testing on macOS the entire time.
+          */}
+          { label ? (
+            <AccessibleValue
+              value={a11yString}
+              readThis={intl.formatMessage({
+                id: 'Terra.timeInput.labeledTextValue',
+                // eslint-disable-next-line formatjs/no-camel-case
+                defaultMessage: '{label} {a11yString}',
+                description: `This will be read to screen reader users only textValue changes to a new time. We want
+              to give the screen reader user feedback that their change to one of the controls has updated this time.`,
+              }, { label, a11yString })}
+            />
+          ) : (
+            <AccessibleValue
+              value={a11yString}
+              readThis={intl.formatMessage({ // fix
+                id: 'Terra.timeInput.textValue',
+                // eslint-disable-next-line formatjs/no-camel-case
+                defaultMessage: 'Time {a11yString}',
+                description: `Similar to Terra.timeInput.labeledTextValue, but we want the screen reader to say "time"
+                before reading the value because no label was provided. It would be confusing to hear "09 22" right
+                after you typed "22" or "09". So instead we can say "Time 09 22".`,
+              }, { a11yString })}
+            />
+          )}
```

Comment:
It seems like this is creating the same AccessibleValue but with a different "readThis" depending on whether or not a label is provided. Is it possible to consolidate these into one AccessibleValue by creating the valid label before?

`const a11yLabel = label || intl.formatMessage({ id: 'Terra.timeInput.textValue' })`

After we have a11yLabel that we know is not empty, then create AccessibleValue without the duplication?

Reply:
I'll try it.